plugin-fancymenu/lxqtfancymenuwindow.cpp

```diff
@@ -0,0 +1,514 @@
+ * Public License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * END_COMMON_COPYRIGHT_HEADER */
+
+
+#include "lxqtfancymenuwindow.h"
+
+#include "lxqtfancymenuappmap.h"
+#include "lxqtfancymenuappmodel.h"
+#include "lxqtfancymenucategoriesmodel.h"
+
+#include <QLineEdit>
+#include <QToolButton>
+#include <QListView>
+
+#include <QMenu>
+#include <QStandardPaths>
+#include <QDir>
+#include <QMimeData>
+#include <XdgIcon>
+#include <QFile>
+
+#include <QApplication>
+#include <QClipboard>
+
+#include <QBoxLayout>
+
+#include <QMessageBox>
+
+#include <QProcess>
+
+#include <QKeyEvent>
+#include <QCoreApplication>
+
+#include <QProxyStyle>
+#include <QStyledItemDelegate>
+
+namespace
+{
+class SingleActivateStyle : public QProxyStyle
+{
+public:
+    using QProxyStyle::QProxyStyle;
+    int styleHint(StyleHint hint, const QStyleOption * option = nullptr, const QWidget * widget = nullptr, QStyleHintReturn * returnData = nullptr) const override
+    {
+        if(hint == QStyle::SH_ItemView_ActivateItemOnSingleClick)
+            return 1;
+        return QProxyStyle::styleHint(hint, option, widget, returnData);
+
+    }
+};
+
+class SeparatorDelegate : public QStyledItemDelegate
+{
+public:
+    SeparatorDelegate(QObject *parent) : QStyledItemDelegate(parent) {}
+
+    static bool isSeparator(const QModelIndex &index)
+    {
+        return index.data(LXQtFancyMenuItemIsSeparatorRole).toInt() == 1;
+    }
+
+protected:
+    void paint(QPainter *painter,
+               const QStyleOptionViewItem &option,
+               const QModelIndex &index) const override
+    {
+        if (isSeparator(index))
+        {
+            QRect rect = option.rect;
+            if (const QAbstractItemView *view = qobject_cast<const QAbstractItemView*>(option.widget))
+                rect.setWidth(view->viewport()->width());
+            QStyleOption opt;
+            opt.rect = rect;
+            option.widget->style()->drawPrimitive(QStyle::PE_IndicatorToolBarSeparator, &opt, painter, option.widget);
+        }
+        else
+        {
+            QStyledItemDelegate::paint(painter, option, index);
+        }
+    }
+
+    QSize sizeHint(const QStyleOptionViewItem &option,
+                   const QModelIndex &index) const override
+    {
+        if (isSeparator(index))
+        {
+            int pm = option.widget->style()->pixelMetric(QStyle::PM_DefaultFrameWidth,
+                                                         nullptr,
+                                                         option.widget);
+            return QSize(pm, pm);
+        }
+
+        return QStyledItemDelegate::sizeHint(option, index);
+    }
+};
+
+}
+
+LXQtFancyMenuWindow::LXQtFancyMenuWindow(QWidget *parent)
+    : QWidget{parent, Qt::Popup}
+{
+    SingleActivateStyle *s = new SingleActivateStyle(style());
+    s->setParent(this);
```

Comment:
When I took a look at the code, by chance I found the above line. It is the cause of both crashes mentioned in https://github.com/lxqt/lxqt-panel/pull/1975#issuecomment-1890615734.